### PR TITLE
feat(argocd-apps): finalizers for applicationset

### LINKED ISSUE #2951: Allows finalizers to be set on Applicationset via argocd-apps helm chart
### Is your feature request related to a problem?

Currently it is not possible to set finalizer on applicationsets via argocd-apps helm chart. This is desirable for below
https://argo-cd.readthedocs.io/en/stable/operator-manual/applicationset/Controlling-Resource-Modification/#how-to-prevent-application-controller-from-deleting-applications-when-deleting-applicationset

### Related helm chart

argocd-apps

### Describe the solution you'd like

set applicationset finalizer via values.

### Describe alternatives you've considered

Set the finalizers manually after applicationsets have been created.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by mannbiher**
Resolves #2951 

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argocd-apps/Chart.yaml [hunk 1]
```diff
@@ -19,3 +19,6 @@ annotations:
   artifacthub.io/changes: |
     - kind: fixed
       description: not rendering empty app description
+  artifacthub.io/changes: |
+    - kind: added
+      description: allow setting applicationset finalizers
```
Comment (mkilchhofer) [inline charts/argocd-apps/Chart.yaml:24]: line 19..21 needs to be removed

**Review by mkilchhofer — COMMENTED** [review]
Almost LGTM.
Can you please remove the changelog lines from the previous release?

File: charts/argocd-apps/Chart.yaml (hunk 1 above)
Comment (mannbiher) [reply charts/argocd-apps/Chart.yaml]: corrected.

Comment (mannbiher) [reply charts/argocd-apps/Chart.yaml]: Restored wrong deletion. Now the Chart changelog reflects current release changes.

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. LGTM.

**Review by mkilchhofer — APPROVED** [review]

**Pull request merged**